src/app/zap-templates/zcl/data-model/chip/general-commissioning-cluster.xml

```diff
@@ -35,8 +35,10 @@ limitations under the License.
     <code>0x0030</code>
     <define>GENERAL_COMMISSIONING_CLUSTER</define>
     <description>This cluster is used to set, remove and update fabric information on a commissionee.</description>
-    <attribute side="server" code="0x00" define="FABRIC_ID" type="OCTET_STRING" length="8" writable="false" default="o" optional="false">FabricId</attribute>
-    <attribute side="server" code="0x01" define="BREADCRUMB" type="INT64U" min="0x0000000000000000" max="0xFFFFFFFFFFFFFFFF" writable="true" optional="false">Breadcrumb</attribute>
+    <attribute side="server" code="0x00" define="BREADCRUMB" type="INT64U" min="0x0000000000000000" max="0xFFFFFFFFFFFFFFFF" writable="true" default="0x0000000000000000" optional="false">Breadcrumb</attribute>
+    <attribute side="server" code="0x01" define="BASICCOMMISSIONINGINFO_LIST" type="ARRAY" entryType="BasicCommissioningInfoType" length="254" writable="false" optional="false">BasicCommissioningInfoList</attribute>
+    <attribute side="server" code="0x02" define="REGULATORYCONFIG_LIST" type="ARRAY" entryType="RegulatoryLocationType" length="254" writable="false" optional="true">RegulatoryConfigList</attribute>
+    <attribute side="server" code="0x03" define="LOCATIONCAPABILITY_LIST" type="ARRAY" entryType="RegulatoryLocationType" length="254" writable="false" optional="true">LocationCapabilityList</attribute>
```

Comment:
The spec mentions a default value of `Indoor/Outdoor` for LocationCapability. Do we have a limitation to assign default to structs/array ? 

Also want to add, I know `length="254"` as been used in most case where we use an type(array) + entryType(struct) due to the limitation of generating a type Struct directly.
But it is more exact to use  a lenght of `2 + (arraysize(1) * Size of Struct (X))` as we only want 1 RegulatoryConfig and 1 LocationCapability and that is a lot less than 254

Reply:
That is correct, this is known issue. https://github.com/project-chip/zap/issues/151, we need to provide ability to specify complex default values and ability for the UI to edit them.

All 'ARRAY' currently use length 254, I raised this question in chat and I remember I was told this is maximum allowed size, not actual allocated size, is not true?

Reply:
Ok, I wasn't sure of that, I recall a conversation I had with Boris @bzbarsky-apple  about those length values. I know the size need to be big enough to accomodate the struct address and struct size but I am not completely sure if all the 254 bytes are allocated.